diff --git a/node.gyp b/node.gyp
index ec1f90b7..0d672e32 100644
--- a/node.gyp
+++ b/node.gyp
@@ -61,6 +61,7 @@
       '<@(node_builtin_shareable_builtins)',
     ],
     'node_sources': [
+      'src/node_embedding_api.cc',
       'src/api/async_resource.cc',
       'src/api/callback.cc',
       'src/api/embed_helpers.cc',
diff --git a/src/node_embedding_api.cc b/src/node_embedding_api.cc
new file mode 100644
index 00000000..3d6f0bc6
--- /dev/null
+++ b/src/node_embedding_api.cc
@@ -0,0 +1,20 @@
+//
+// Description: C-based API for embedding Node.js.
+//
+// !!! WARNING !!! WARNING !!! WARNING !!!
+// This is a new API and is subject to change.
+// While it is C-based, it is not ABI safe yet.
+// Consider all functions and data structures as experimental.
+// !!! WARNING !!! WARNING !!! WARNING !!!
+//
+// This file contains the C-based API for embedding Node.js in a host
+// application. The API is designed to be used by applications that want to
+// embed Node.js as a shared library (.so or .dll) and can interop with
+// C-based API.
+//
+
+#include "node.h"
+
+extern "C" int node_embedding_start(int argc, char** argv) {
+  return node::Start(argc, argv);
+}
diff --git a/src/node_embedding_api.h b/src/node_embedding_api.h
new file mode 100644
index 00000000..17e080bc
--- /dev/null
+++ b/src/node_embedding_api.h
@@ -0,0 +1,18 @@
+//
+// Description: C-based API for embedding Node.js.
+//
+// !!! WARNING !!! WARNING !!! WARNING !!!
+// This is a new API and is subject to change.
+// While it is C-based, it is not ABI safe yet.
+// Consider all functions and data structures as experimental.
+// !!! WARNING !!! WARNING !!! WARNING !!!
+//
+// This file contains the C-based API for embedding Node.js in a host
+// application. The API is designed to be used by applications that want to
+// embed Node.js as a shared library (.so or .dll) and can interop with
+// C-based API.
+//
+
+#include "node.h"
+
+int node_embedding_start(int argc, char** argv);
